Sources/SwiftGenKit/Parsers/Fonts/FontsParser.swift

```diff
@@ -35,18 +35,41 @@ public enum Fonts {
 
 // MARK: - Helpers
 
+private extension Data {
+    var isTrueTypeFont: Bool {
+        var values = [UInt8](repeating: 0, count:5)
+        self.copyBytes(to: &values, count: 5)
+        let firstBytesOfTTF: [UInt8] = [0x00, 0x01, 0x00, 0x00, 0x00] // https://en.wikipedia.org/wiki/List_of_file_signatures
+        return values == firstBytesOfTTF
+    }
+
+    var isOpenTypeFont: Bool {
+        var values = [UInt8](repeating: 0, count:4)
+        self.copyBytes(to: &values, count: 4)
+        let firstBytesOfOTF: [UInt8] = [0x4F, 0x54, 0x54, 0x4F] // https://en.wikipedia.org/wiki/List_of_file_signatures
+        return values == firstBytesOfOTF
+    }
+}
+
 private extension Fonts.Parser {
   /// Try to quickly check if the given file is an actual font file.
   ///
   /// - Note: This can seemingly fail in sandboxed environments.
   func checkIsFont(path: Path) -> Bool {
-    guard let values = try? path.url.resourceValues(forKeys: [.typeIdentifierKey]),
-      let uti = values.typeIdentifier else {
-      warningHandler?("Unable to determine the Universal Type Identifier for file \(path)", #file, #line)
+    do {
+      let values = try path.url.resourceValues(forKeys: [.typeIdentifierKey])
+      guard let uti = values.typeIdentifier else {
+        return true
+      }
+      return UTTypeConformsTo(uti as CFString, kUTTypeFont)
+    } catch {
+      if let data = try? Data(contentsOf: path.url),
```

Comment:
This will load the whole file into memory, which is (or can be) slow. Instead, we should use [FileHandle.read(upToCount:)](https://developer.apple.com/documentation/foundation/filehandle/3516317-read), with a maximum of N header bytes.

Note: best to create a constant for the amount of header bytes, just to make it clear.